lib-gphotos/client.go

```diff
@@ -102,29 +104,49 @@ func (client *Client) UploadFile(filePath string, pAlbumID ...string) (*photosli
 		return nil, stacktrace.Propagate(err, "failed getting uploadToken for %s", filename)
 	}
 
-	batchResponse, err := client.MediaItems.BatchCreate(&photoslibrary.BatchCreateMediaItemsRequest{
-		AlbumId: albumID,
-		NewMediaItems: []*photoslibrary.NewMediaItem{
-			&photoslibrary.NewMediaItem{
-				Description:     filename,
-				SimpleMediaItem: &photoslibrary.SimpleMediaItem{UploadToken: uploadToken},
+	retry := true
+	retryCount := 0
+	for retry != false {
+		retry = false
+		batchResponse, err := client.MediaItems.BatchCreate(&photoslibrary.BatchCreateMediaItemsRequest{
+			AlbumId: albumID,
+			NewMediaItems: []*photoslibrary.NewMediaItem{
+				&photoslibrary.NewMediaItem{
+					Description:     filename,
+					SimpleMediaItem: &photoslibrary.SimpleMediaItem{UploadToken: uploadToken},
+				},
 			},
-		},
-	}).Do()
-	if err != nil {
-		return nil, stacktrace.Propagate(err, "failed adding media %s", filename)
-	}
+		}).Do()
+		if err != nil {
+			// handle rate limit error by sleeping and retrying
+			if strings.HasPrefix(err.Error(), "googleapi: Error 429") {
+				log.Printf("Rate limit reached, sleeping for 10 seconds...")
+				time.Sleep(10 * time.Second)
```

Comment:
When a Server sends a 429, It usually add a `Retry-After`  header:
```
after, err := strconv.ParseInt(batchResponse.ServerResponse.Header.Get("Retry-After"), 10, 64)
if err != nil || after == 0 { after = 10 }
time.Sleep(after * time.Second)
```

Reply:
thanks. FYI you need to convert int64 after to a time.Duration.